src/wp-includes/meta.php

```diff
@@ -1496,3 +1593,69 @@ function get_object_subtype( $object_type, $object_id ) {
 	 */
 	return apply_filters( "get_object_subtype_{$object_type}", $object_subtype, $object_id );
 }
+
+/**
+ * Filter into default_{$object_type}_metadata and add in default value.
+ *
+ * @since 5.5.0
+ *
+ * @param mixed  $value     Current value passed to filter.
+ * @param string $meta_type Type of object metadata is for (e.g., comment, post, term, or user).
+
+ * @param string $meta_key  Optional. Metadata key. If not specified, retrieve all metadata for
+ *                          the specified object.
+ * @param bool   $single    Optional, default is false.
+ *                          If true, return only the first value of the specified meta_key.
+ *                          This parameter has no effect if meta_key is not specified.
+ * @param int    $object_id ID of the object metadata is for
+ *
+ * @return mixed Single metadata default, or array of defaults
+ */
+function filter_default_metadata( $value, $meta_type, $meta_key, $single, $object_id ) {
+	global $wp_meta_keys;
+
+	if ( wp_installing() ) {
+		return $value;
+	}
+
+	if ( ! is_array( $wp_meta_keys ) || ! isset( $wp_meta_keys[ $meta_type ] ) ) {
+		return $value;
+	}
+
+	$defaults = array();
+	foreach ( $wp_meta_keys[ $meta_type ] as $sub_type => $meta_data ) {
+		foreach ( $meta_data as $_meta_key => $args ) {
+			if ( $_meta_key === $meta_key && array_key_exists( 'default', $args ) ) {
+				$defaults[ $sub_type ] = $args;
+			}
+		}
+	}
+
+	if ( ! $defaults ) {
+		return $value;
+	}
+
+	if ( isset( $defaults[''] ) ) {
```

Comment:
I'm not following this check. What is happening here?

Reply:
I build an array of registered default values, that looks like this 
```
$defaults[''] = [ 'key' => 'testing', 'default' => 'goodbye'];
$defaults['post'] = [ 'key' => 'testing', 'default' => 'hello' ];
```

If a subtype is not set, then the key is an empty string.

Reply:
Thanks for the clarification. I've added a comment describing that.